spec/handlers/newflow/send_reset_password_email_spec.rb

```diff
@@ -54,5 +54,23 @@ module Newflow
         expect(result).to have_routine_error(:cannot_find_user)
       end
     end
+
+    context 'when no email present and no logged in user' do
+      let(:params) do
+        {
+          forgot_password_form: {
+            email: ''
```

Comment:
is no email present empty string AND nil?  email: nil?  having '' here is unclear to this.

Reply:
**I believe** Rails `params` never post as `nil` for empty fields. Instead, they get submitted or interpreted as empty strings. Am I wrong? 🤔

Reply:
i ask because it's prob. important to understand the possible test cases here:  no email field, nil email field, empty string email field, for thoroughness of the test.  just asking if you've done that here.

Reply:
I am not going to test all three cases (no email field, nil email field, empty string email field), I only whatever would get submitted by a form with nothing in the email field (in other words, when an end-user leaves the email field empty and submits the form that way... which I believe becomes an empty string in Rails in the `params` object.

Reply:
Either way, both empty string and nil evaluate to false by the `.present?` method which is what's important here. 

```
...
        unless outputs.email.present? || logged_in_user
```